posthog/models/filters/retention_filter.py

```diff
@@ -43,27 +45,57 @@ def __init__(self, data: Optional[Dict[str, Any]] = None, request: Optional[Http
         self.retention_type = data.get(RETENTION_TYPE, self.retention_type)
         self.total_intervals = data.get(TOTAL_INTERVALS, self.total_intervals)
         self.selected_interval = int(data.get(SELECTED_INTERVAL, 0))
+        tdelta, t1 = RetentionFilter.determine_time_delta(self.total_intervals, self.period)
 
-        if not self.date_from:
-            self._date_from = "-11d"
+        self.period_increment = t1
+        self.total_increment = tdelta
 
-        tdelta, t1 = RetentionFilter.determine_time_delta(self.total_intervals, self.period)
-        self._date_to = (self.date_to + t1).isoformat()
+    @property
+    def date_from(self):
+        _date_from = super().date_from
+        date_to = self.date_to
+        if not _date_from:
+            _date_from = relative_date_parse("-11d")
+        _date_from = date_to - self.total_increment
+
+        if self.period == "Week":
+            _date_from = _date_from - timedelta(days=_date_from.isoweekday() % 7)  # type: ignore
+
+        return _date_from
+
```

Comment:
Had to fix date handling in retention filter so that when the celery tasks create the filter, the dates were the same as the initial filter